client/src/components/Category/index.js

```diff
@@ -170,91 +144,132 @@ class Category extends Component {
               <CharityCount refresh={refresh} />
               <div className="container">
                 <div className="result-cards">
-                  {data.slice(0, 5).map(item => {
-                    const {
-                      idChirty,
-                      id,
-                      isActive,
-                      logo,
-                      name,
-                      text,
-                      governance,
-                      impact,
-                      financial,
-                    } = item;
-                    return (
-                      <ResultCard
-                        extraClass="result-cards-caregory"
-                        idChirty={idChirty}
-                        key0={id}
-                        isActive={isActive}
-                        onClickCompare={() =>
-                          this.changeActive(item.id, item.idChirty)
+                  <CountContext.Consumer>
+                    {consumer => {
+                      const { charityList, changeActive } = consumer;
+                      const charityListId = charityList.map(
+                        charity => charity.charityId
+                      );
+                      const dataActive = data.map(charity => {
+                        if (charityListId.includes(charity.idChirty)) {
+                          charity.isActive = true;
+                        } else {
+                          charity.isActive = false;
```

Comment:
@asmaa-ezz charityListId.includes returns true or false, so if you want to save a few lines of code, you could rewrite as `charity.isActive = charityListId.includes(charity.idChirty)`.
It's not an important change to make and is more of a personal preference (whether you like the shorter code or you like the if-statement), but I think the shorter code can help readability sometimes if it's inside a very long function (especially a long render function).